src/hooks/useEvent.ts

```diff
@@ -8,29 +8,38 @@ const useEvent = (id?: string) => {
   const [error, setError] = useState<IError | null | undefined>()
   const [loading, setLoading] = useState<boolean>(false)
 
-  useEffect(() => {
-    let ignore = false
-    if (id) {
+  const handleGetEvent = useCallback(
+    (ignore: boolean) => {
       setLoading(true)
+      if (id) {
+        getEvent(id)
+          .then(({ response }) => {
+            if (!ignore) setEvent(response.data)
+          })
+          .catch((error: IError) => {
+            setError(error)
+          })
+          .finally(() => {
+            setLoading(false)
+          })
+      }
+    },
+    [id],
+  )
 
-      getEvent(id)
-        .then(({ response }) => {
-          if (!ignore) setEvent(response.data)
-        })
-        .catch((error: IError) => {
-          setError(error)
-        })
-        .finally(() => {
-          setLoading(false)
-        })
-    }
+  const reload = useCallback(() => {
+    handleGetEvent(false)
+  }, [handleGetEvent])
```

Comment:
Not sure using a useCallback function as a dependancy for the useCallback is a good standard, can we use the `id` same as the useCallback above?

Reply:
Good catch! I only added handleGetEvent as a dependency cos the linter complains. Any ideas how else I can get around it?

Reply:
Actually having read up I think this is fine!